### PR TITLE
ECOM-6217 Added logs and try catch on order creation

### PR DESCRIPTION
**Pull request by Ayub-Khan**
ECOM-6217

**Description :**
Added try catch around the order.save and logged exception to check if it faces any exception on order.save and cause Missing Order issue. 

**Reviewers :** 
- [ ]  @noraiz-anwar 
- [ ]  @Qubad786

### CODE REVIEW

File: ecommerce/extensions/order/utils.py [hunk 1]
```diff
@@ -99,8 +99,11 @@ def create_order_model(self, user, basket, shipping_address, shipping_method, sh
             order_data['status'] = status
         if extra_order_fields:
             order_data.update(extra_order_fields)
-        order = Order(**order_data)
-        order.save()
+        try:
+            order = Order(**order_data)
+            order.save()
+        except Exception:
```
Comment (noraiz-anwar) [inline ecommerce/extensions/order/utils.py]: I don't think quality test will pass here. It is too broad exception handling and quality check script don't like it.

**Review by noraiz-anwar — DISMISSED** [review]

**Review by noraiz-anwar — COMMENTED** [review]
Need to add tests

**Comment by schenedx** [comment]
@clintonb @rlucioni Please review. This will have an impact to the splunk alerts as well

**Review by schenedx — APPROVED** [review]
👍

File: ecommerce/extensions/order/utils.py [hunk 2]
```diff
@@ -99,8 +99,11 @@ def create_order_model(self, user, basket, shipping_address, shipping_method, sh
             order_data['status'] = status
         if extra_order_fields:
             order_data.update(extra_order_fields)
-        order = Order(**order_data)
-        order.save()
+        try:
+            order = Order(**order_data)
+            order.save()
+        except Exception:  # pylint: disable=broad-except
```
Comment (clintonb) [inline ecommerce/extensions/order/utils.py:105]: You don't need `Exception` here.

**Review by clintonb — APPROVED** [review]

File: ecommerce/extensions/order/utils.py [hunk 3]
```diff
@@ -99,8 +99,11 @@ def create_order_model(self, user, basket, shipping_address, shipping_method, sh
...
+            logger.exception('Failed to save Order [%s]', order_number)
```
Comment (rlucioni) [inline ecommerce/extensions/order/utils.py:106]: Should the exception be re-raised here? Can this method safely continue if saving the order fails?

Comment (clintonb) [reply ecommerce/extensions/order/utils.py:106]: Yes, the exception should be raised. Now that I think about it, any exceptions that previously occurred should already be logged...unless they are caught, squashed, and not logged.

Comment (rlucioni) [reply ecommerce/extensions/order/utils.py:106]: Is the goal of this work simply to log something when orders fail to save? If that's the case, to @clintonb's point, this PR seems unnecessary. It only makes sense if we want this method to proceed in spite of the order being unsaved.

**Review by clintonb — CHANGES_REQUESTED** [review]
Rescinding approval until we determine the goals and necessity of this logging.

File: ecommerce/extensions/order/utils.py (hunk 3 above)
Comment (adampalay) [reply ecommerce/extensions/order/utils.py:106]: We wanted to add more nuanced logging to help us get to the root cause; it seems to me that adding `raise` after we log the exception is the way to go. @rlucioni , the PR is necessary to get that extra information and to make it easier to find in splunk.

Comment (rlucioni) [reply ecommerce/extensions/order/utils.py:106]: @adampalay there should already be stacktraces being logged to Splunk. I'm not convinced this change adds any value. It should already be easy to find instances of the exception in the logs.

Comment (Ayub-Khan) [reply ecommerce/extensions/order/utils.py:106]: @adampalay  Sorry my mistake it's not going to help us in any way. @rlucioni  is right if there is exception its already logged and we can check it using file name search.

Comment (adampalay) [reply ecommerce/extensions/order/utils.py:106]: I see. What about having the order number in the logs? Would that be useful?

Comment (Ayub-Khan) [reply ecommerce/extensions/order/utils.py:106]: @adampalay No its same.

**Pull request closed without merging**